### PR TITLE
Simplify ConnectionListener.RunAsync

### PR DESCRIPTION
**Pull request by ReubenBond**
Fixing in response to feedback from @davidfowl

### CODE REVIEW

File: src/Orleans.Runtime/Networking/ConnectionListener.cs [hunk 1]
```diff
@@ -81,21 +81,11 @@ public void Start()
         private async Task RunAsync()
         {
             var runCancellation = this.cancellation.Token;
-            var runCancellationTask = runCancellation.WhenCancelled();
             while (!runCancellation.IsCancellationRequested)
```
Comment (davidfowl) [inline src/Orleans.Runtime/Networking/ConnectionListener.cs]: while (true), let the AcceptAsync call throw.

**Comment by sergeybykov** [comment]
Functional test consistently get stuck with this change.

**Comment by davidfowl** [comment]
My guess would be that unbind isn’t being used to yield the accept loop. The token isn’t respected by the socket transport

**Comment by ReubenBond** [comment]
I've fixed it in the latest commit and aligned the style with Kestrel's

**Pull request merged**